test/Microsoft.SqlTools.ServiceLayer.IntegrationTests/ObjectManagement/ObjectManagementServiceTests.cs

```diff
@@ -0,0 +1,171 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.SqlServer.Management.Smo;
+using Microsoft.SqlTools.BatchParser.Utility;
+using Microsoft.SqlTools.Hosting.Protocol;
+using Microsoft.SqlTools.ServiceLayer.Connection;
+using Microsoft.SqlTools.ServiceLayer.IntegrationTests.Utility;
+using Microsoft.SqlTools.ServiceLayer.ObjectManagement;
+using Microsoft.SqlTools.ServiceLayer.ObjectManagement.Contracts;
+using Microsoft.SqlTools.ServiceLayer.Scripting;
+using Microsoft.SqlTools.ServiceLayer.Scripting.Contracts;
+using Microsoft.SqlTools.ServiceLayer.Test.Common;
+using Moq;
+using NUnit.Framework;
+using static Microsoft.SqlTools.ServiceLayer.IntegrationTests.Utility.LiveConnectionHelper;
+
+namespace Microsoft.SqlTools.ServiceLayer.IntegrationTests.ObjectManagement
+{
+    public class ObjectManagementServiceTests
+    {
+        private readonly string table_query = @"CREATE TABLE testTable1_RenamingTable (c1 int)
+                            GO
+                            ";
+        private readonly string ownerUri = "testDB";
+        private readonly string startTableName = "testTable1_RenamingTable";
+        private readonly string startColumnName = "c1";
+        private ObjectManagementService objectManagementService;
+        private SqlTestDb testDb;
+        private Mock<RequestContext<bool>> requestContextMock;
+
+        [SetUp]
+        public async Task TestInitialize()
+        {
+            this.testDb = await SqlTestDb.CreateNewAsync(TestServerType.OnPrem, false, null, table_query, "RenameTest");
+
+            requestContextMock = new Mock<RequestContext<bool>>();
+            ConnectionService connectionService = LiveConnectionHelper.GetLiveTestConnectionService();
+
+            TestConnectionResult connectionResult = await LiveConnectionHelper.InitLiveConnectionInfoAsync(testDb.DatabaseName, this.ownerUri, ConnectionType.Default);
+
+            ObjectManagementService.ConnectionServiceInstance = connectionService;
+            this.objectManagementService = new ObjectManagementService();
+        }
+
+        [Test]
+        public async Task TestRenameTable()
+        {
+            await objectManagementService.HandleRenameRequest(this.InitRequestParams("RenamingTable", String.Format("Server/Database[@Name='{0}']/Table[@Name='testTable1_RenamingTable' and @Schema='dbo']", testDb.DatabaseName)), requestContextMock.Object);
+            Thread.Sleep(2000);
+            requestContextMock.Verify(x => x.SendResult(It.Is<bool>(r => VerifySendedFeedback(r, true))));
+            await CheckGeneratedScript(testDb.DatabaseName, "CREATE TABLE [dbo].[RenamingTable]");
+
+            await SqlTestDb.DropDatabase(testDb.DatabaseName);
```

Comment:
The drop should be done as part of a teardown function so that a failure in the test doesn't cause it to skip cleanup